contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -320,112 +304,121 @@ contract MixinStakingPoolRewards is
+        if (unsyncedStake.currentEpoch == _currentEpoch) {
             return 0;
         }
 
-        // If there are unfinalized rewards this epoch, compute the member's
-        // share.
-        if (unfinalizedMembersReward != 0 && unfinalizedMembersStake != 0) {
-            // Unfinalized rewards are always earned from stake in
-            // the prior epoch so we want the stake at `currentEpoch-1`.
-            uint256 _stake = unsyncedStake.currentEpoch >= currentEpoch.safeSub(1) ?
-                unsyncedStake.currentEpochBalance :
-                unsyncedStake.nextEpochBalance;
-            if (_stake != 0) {
-                reward = LibMath.getPartialAmountFloor(
-                    unfinalizedMembersReward,
-                    unfinalizedMembersStake,
-                    _stake
-                );
-            }
-        }
-
-        // Get the last epoch where a reward was credited to this pool, which
-        // also happens to be when we last created a cumulative reward entry.
-        uint256 lastRewardEpoch = _cumulativeRewardsByPoolLastStored[poolId];
+        // We account for rewards over 3 intervals, below.
 
-        // If the stake has been touched since the last reward epoch,
-        // it has already been claimed.
-        if (unsyncedStake.currentEpoch >= lastRewardEpoch) {
-            return reward;
-        }
-        // From here we know: `unsyncedStake.currentEpoch < currentEpoch > 0`.
+        // 1/3 Unfinalized rewards earned in `currentEpoch - 1`.
+        reward = _computeUnfinalizedDelegatorReward(
+            unsyncedStake,
+            _currentEpoch,
+            unfinalizedMembersReward,
+            unfinalizedMembersStake
+        );
 
-        uint256 nextStakeEpoch = uint256(unsyncedStake.currentEpoch).safeAdd(1);
+        // 2/3 Finalized rewards earned in epochs [`unsyncedStake.currentEpoch + 1` .. `currentEpoch - 1`]
+        uint256 unsyncedStakeNextEpoch = uint256(unsyncedStake.currentEpoch).safeAdd(1);
         reward = reward.safeAdd(
             _computeMemberRewardOverInterval(
                 poolId,
                 unsyncedStake.currentEpochBalance,
                 unsyncedStake.currentEpoch,
-                nextStakeEpoch
+                unsyncedStakeNextEpoch
             )
         );
-        if (nextStakeEpoch < lastRewardEpoch) {
-            reward = reward.safeAdd(
-                _computeMemberRewardOverInterval(
-                    poolId,
-                    unsyncedStake.nextEpochBalance,
-                    nextStakeEpoch,
-                    lastRewardEpoch
-                )
-            );
-        }
+
+        // 3/3 Finalized rewards earned in epoch `unsyncedStake.currentEpoch`.
+        reward = reward.safeAdd(
+            _computeMemberRewardOverInterval(
+                poolId,
+                unsyncedStake.nextEpochBalance,
+                unsyncedStakeNextEpoch,
+                _currentEpoch
```

Comment:
Hmm, doesn't this miss rewards that have been credited an to a pool AFTER a delegator last synced? Also, isn't it possible that neither `_currentEpoch` nor `_currentEpoch-1` would have entries so this could fail?

Reply:
discussed offline and I'm wrong :upside_down_face: